modules/hal_gigadevice/Kconfig

```diff
@@ -52,6 +52,35 @@ config GD32_HXTAL_25MHZ
 
 endchoice
 
+choice GD32_LOW_SPEED_IRC_FREQUENCY
+	prompt "Internal RC oscillator clock frequency"
+	default GD32_LOW_SPEED_IRC_32K if SOC_SERIES_GD32F4XX
+	default GD32_LOW_SPEED_IRC_40K if !SOC_SERIES_GD32F4XX
+	help
+	  Define value of low speed internal RC oscillator (IRC) in Hz
+	  This value sets the frequency of the oscillator.
+
+config GD32_LOW_SPEED_IRC_32K
+	bool "32KHz"
+	depends on SOC_SERIES_GD32F4XX
+	help
+	  Use 32KHz oscillator for low speed internal RC Oscillator
+
+config GD32_LOW_SPEED_IRC_40K
+	bool "40KHz"
+	depends on !SOC_SERIES_GD32F4XX
+	help
+	  Use 40KHz oscillator for low speed internal RC Oscillator
+
+endchoice
+
+config GD32_LOW_SPEED_IRC_HZ
+	int
+	default 32000 if GD32_LOW_SPEED_IRC_32K
+	default 40000 if GD32_LOW_SPEED_IRC_40K
+	help
+	  Numeric value of low speed IRC frequency (in HZ)
+
```

Comment:
There is no need to define this choice, just keep below snip and adjust name at GD32_LOW_SPEED_IRC_HZ.

```
config GD32_HAS_IRC_32K
	bool
	help
	  Use 32KHz oscillator for low speed internal RC Oscillator

config GD32_HAS_IRC_40K
	bool
	help
	  Use 40KHz oscillator for low speed internal RC Oscillator
```

Then add on each SoC Kconfig.series the correct option. The **GD32_HAS_**_FOO_ should be defined on each SoC to keep Kconfig and CMake rules clean with more granularity.

```
 config SOC_SERIES_GD32F4XX 
 	bool "GigaDevice GD32F4XX series Cortex-M4F MCU" 
 	select ARM 
 	select CPU_HAS_ARM_MPU 
 	select CPU_HAS_FPU 
 	select CPU_CORTEX_M4 
 	select SOC_FAMILY_GD32_ARM 
 	select GD32_HAS_AF_PINMUX 
+ 	select GD32_HAS_IRC_32K
```

Reply:
Thanks for reviewing. 
And sorry for my late response.

```
+config GD32_DBG_SUPPORT
+	bool "Enable GD32 Debug features"
+	select USE_GD32_DBG
+	depends on DEBUG
# or even
+	default y if DEBUG
...
	bool "GD32 Free watchdog timer (FWDGT) driver"
	default $(dt_compat_enabled,$(DT_COMPAT_GIGADEVICE_GD32_FWDGT))
	select USE_GD32_FWDGT
-	select USE_GD32_DBG
```

I got it.
I apply your suggestion with using  `default y if DEBUG`

Reply:
> Then add on each SoC Kconfig.series the correct option. The GD32_HAS_FOO should be defined on each SoC to keep Kconfig and CMake rules clean with more granularity.

I understood it. And apply it.
By the way, I can only use GD32VF103 to verify this change.
I modified the source related to it.
Should I better apply this change to all other GD32-SoCs?